src/libraries/Microsoft.PowerFx.Connectors/PowerPlatformConnectorClient.cs

```diff
@@ -18,7 +19,32 @@ public class PowerPlatformConnectorClient : HttpClient
         /// <summary>
         /// For telemetry - assembly version stamp. 
         /// </summary>
-        public static string Version { get; } = typeof(PowerPlatformConnectorClient).Assembly.GetName().Version.ToString();
+        public static string Version
+        {
+            get
+            {
+                if (!string.IsNullOrEmpty(_version))
+                {
+                    return _version;
+                }                
+
+                try
+                {
+                    // Will return "0.2.6-preview.20230615-1002"
+                    // ProductVersion example "0.2.6-preview.20230615-1002+048250dc4404bdb749f992cd0869f8821c79e3fe"
+                    _version = FileVersionInfo.GetVersionInfo(typeof(PowerPlatformConnectorClient).Assembly.Location).ProductVersion.Split('+')[0];
+                }
+                catch (Exception)
```

Comment:
Here we are accessing the file system and trying to read the DLL, there is no guarantee we'll be allowed to do that from a security standpoint.
I prefer this extra safety net, just in case...